### PR TITLE
Add retrier

### PR DESCRIPTION
**Pull request by micahhausler**
* Add Makefile
* Add .gitignore
* Add tests to retrier operation

Signed-off-by: Micah Hausler <hausler.m@gmail.com>

### CODE REVIEW

File: cmd/download.go [hunk 1]
```diff
@@ -50,30 +62,45 @@ var downloadCmd = &cobra.Command{
 			response codes, other times it speaks out of turn. "Fail and retry"
 			is sloppy, and yet here we are. I hate this so much.
 			*/
-			for i := 0; i < 10; i++ {
-
-				resp, err := grab.Get(output, source)
+			err = backoff.Retry(
+				GetOperation(ctx, output, source),
+				backoff.WithContext(backoff.NewConstantBackOff(retry), ctx))
```
Comment (micahhausler) [inline cmd/download.go:67]: You're probably not dealing with something that is getting a ton of clients, but you could use a [`NewExponentialBackoff()`](https://pkg.go.dev/github.com/cenkalti/backoff/v4#NewExponentialBackOff) if you wanted to

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -16,10 +16,10 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v3
       with:
-        go-version: 1.17
+        go-version: 1.18
```
Comment (micahhausler) [inline .github/workflows/go.yml:19]: The `go.mod` specified 1.18 so I updated this to match

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -16,10 +16,10 @@ jobs:
...
 
     - name: Build
-      run: go build -v ./...
+      run: make build
```
Comment (micahhausler) [inline .github/workflows/go.yml:22]: I threw everything in a simple `Makefile` as is customary in a lot of Go projects. I can back this out if you don't want it

File: cmd/download.go [hunk 4]
```diff
@@ -50,30 +62,45 @@ var downloadCmd = &cobra.Command{
...
+		fmt.Println("Download saved to", resp.Filename)
+		return nil
+	}
+}
+
 func init() {
 	downloadCmd.Flags().StringP("output", "o", "", "Directory to download into")
 	downloadCmd.Flags().Bool("delete", false, "Delete files after download")
+	downloadCmd.Flags().Duration("retry-interval", time.Second*2, "Time between retries")
+	downloadCmd.Flags().DurationP("timeout", "t", time.Second*20, "Timeout to cancel the command")
```
Comment (micahhausler) [inline cmd/download.go:104]: I have no idea what a reasonable default is, I just picked 20s

File: .github/workflows/go.yml (hunk 3 above)
Comment (quinnypig) [reply .github/workflows/go.yml:22]: I'm so new at this I have no idea what's customary. Thanks!

File: .github/workflows/go.yml (hunk 2 above)
Comment (quinnypig) [reply .github/workflows/go.yml:19]: Yeah, I haven't fully built this out yet. It's on a backlog for "someday."

File: cmd/download.go (hunk 1 above)
Comment (quinnypig) [reply cmd/download.go:67]: If there's a second client on this thing I've been compromised!

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.PHONY: all
+all: build
+
+.PHONY: build
+build:
+	go build -v ./...
```
Comment (quinnypig) [inline Makefile:6]: This doesn't appear to do anything; it presents as a noop. Hmm...

Comment (micahhausler) [reply Makefile:6]: You're right! I kept this the same as what you had in CI, you'll just want to remove the `./...` to get a binary

**Pull request merged**